src/models/AbstractStyledComponent.js

```diff
@@ -12,10 +12,19 @@ export default class AbstractStyledComponent extends Component {
     generatedClassName?: string,
     generatedStyles?: any
   }
-  unsubscribe: () => void
+
+  unsubscribeId: number = -1
+
+  unsubscribeFromContext() {
```

Comment:
Makes sense, the `AbstractStyledComponent` class doesn't seem to provide much utility vs having an extra class in the hierarchy. imho we should try to but as much into the base class as possible.